pkg/signature/kms/kms.go

```diff
@@ -69,14 +67,15 @@ func Get(ctx context.Context, keyResourceID string, hashFunc crypto.Hash, opts .
 			return sv, nil
 		}
 	}
-	if pi, ok := providersMap[CLIPluginProviderKey]; ok {
-		sv, err := pi(ctx, keyResourceID, hashFunc, opts...)
-		if err != nil {
-			return nil, err
-		}
-		return sv, nil
-	}
-	return nil, &ProviderNotFoundError{ref: keyResourceID}
+	// We don't return a ProviderNotFoundError because cosign currently interprets those in a confusing way:
```

Comment:
This library is used outside of Cosign so we shouldn't be making changes just because of how Cosign outputs this error.

Isn't it more accurate to log that it is an unrecognized scheme rather than a missing executable? Whether the KMS provider is an executable or module is a detail the user doesn't need to be aware of.

I think we can get the best of both worlds - wrap the original error from `cliplugin.LoadSignerVerifier` in a `ProviderNotFoundError`.

Reply:
changed, and added tests